src/commands/installExtension.ts

```diff
@@ -3,10 +3,33 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { IActionContext } from "@microsoft/vscode-azext-utils";
-import { commands } from "vscode";
+import { DialogResponses, IActionContext } from "@microsoft/vscode-azext-utils";
+import { commands, MessageItem, window } from "vscode";
+import { getAzureExtensions } from "../AzExtWrapper";
+import { ext } from "../extensionVariables";
+import { localize } from "../utils/localize";
 
-export async function installExtension(_context: IActionContext, extensionId: string): Promise<void> {
-    void commands.executeCommand('extension.open', extensionId);
-    void commands.executeCommand('workbench.extensions.installExtension', extensionId);
+export async function installExtension(context: IActionContext, extensionId: string): Promise<void> {
+    context.telemetry.properties.extensionId = extensionId;
+    await commands.executeCommand('extension.open', extensionId);
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const extension = getAzureExtensions().find(azExt => azExt.id === extensionId)!;
+    const installMessageItem: MessageItem = {
+        title: localize('installExtension', 'Install extension')
+    }
+    // Use extension-specific message later
+    const result = await window.showInformationMessage(localize('installExtension', "Install the '{0}' extension to enable additional features?", extension.label), {
+        modal: true
+    }, installMessageItem, DialogResponses.cancel);
+
+    if (result === installMessageItem) {
+        context.errorHandling.suppressDisplay = true;
+        // Will prompt if settings sync is enabled
+        await commands.executeCommand('workbench.extensions.installExtension', extensionId);
```

Comment:
I'm concerned about what happens when the extension is installed but disabled. AFAIK, VSCode offers no way to see if that is the case, nor a command to enable/disable extensions. If we just directly call the `workbench.extensions.installExtension` it's a noop, which is fine.

Additionally, if it's not installed, they'll be getting two modal popups in a row to install.

I wouldn't want to diminish the user experience to gain telemetry. Likely, VSCode has telemetry we can view for enablement/disablement and install/uninstall, so I'd say we capture the `extensionId` but otherwise just call `workbench.extensions.installExtension` unconditionally.

Reply:
If we set up some way for the extension to activate immediately in this scenario, then the resolver flow will already take care of that--when the extension activates and the resolver registers, it will automatically re-resolve all those nodes. That way we don't need to await the install or refresh the tree.

Just need to change the activation manager to keep track of types it has already fetched/resolved, and if an extension comes in that requests activation based on a type that has already been seen, it gets immediately activated.